dm-manager/pkg/grpc/grpc_server/grpc_server.go

```diff
@@ -0,0 +1,281 @@
+	if hostInv.AmtStatus == "ENABLED" {
+		if hostInv.DesiredAmtState == computev1.AmtState_AMT_STATE_PROVISIONED &&
+			(hostInv.CurrentAmtState == computev1.AmtState_AMT_STATE_UNPROVISIONED ||
+				hostInv.CurrentAmtState == computev1.AmtState_AMT_STATE_UNSPECIFIED) {
+			zlog.Debug().Msgf("Send activation request for Host %s ", req.HostId)
+			response = &pb.ActivationDetailsResponse{
+				Operation:   pb.OperationType_ACTIVATE,
+				HostId:      req.HostId,
+				ProfileName: tenantID,
+			}
+		} else {
+			zlog.Debug().Msgf("Node is provisioned for UUID %s and tID=%s\n",
+				req.HostId, tenantID)
+			return nil, errors.Errorfc(codes.FailedPrecondition,
+				"current state is %s or activation not requested by user %v", hostInv.CurrentAmtState, err)
+		}
+	} else {
+		zlog.Debug().Msgf("AMT is not enabled for host %s", req.HostId)
+		return nil, errors.Errorfc(codes.FailedPrecondition,
+			"AMT is not enabled for host %s", req.HostId)
+	}
+
+	return response, nil
+}
+
+func (dms *DeviceManagementService) ReportActivationResults(
+	ctx context.Context, req *pb.ActivationResultRequest,
+) (*pb.ActivationResultResponse, error) {
+	zlog.Info().Msgf("ReportActivationResults")
+
+	if dms.authEnabled {
+		if !dms.rbac.IsRequestAuthorized(ctx, rbac.CreateKey) {
+			err := errors.Errorfc(codes.PermissionDenied, "Request is blocked by RBAC")
+			zlog.InfraSec().InfraErr(err).Msgf("Request Device management is not authenticated")
+		}
+	}
+
+	tenantID, err := dms.getTenantFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	hostInv, err := dms.getHostByUUID(ctx, tenantID, req.HostId)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: currently activation status is handled by dm-manager
+
+	host := &computev1.HostResource{}
+	if req.ActivationStatus == pb.ActivationStatus_PROVISIONED {
+		host.CurrentAmtState = computev1.AmtState_AMT_STATE_PROVISIONED
+	} else {
+		host.CurrentAmtState = computev1.AmtState_AMT_STATE_UNPROVISIONED
+	}
+
+	if hostInv.CurrentAmtState != host.CurrentAmtState {
+		if req.ActivationStatus == pb.ActivationStatus_PROVISIONED &&
+			hostInv.DesiredAmtState == computev1.AmtState_AMT_STATE_PROVISIONED {
+			zlog.Debug().Msgf("Host %s AMT state changed to provisioned", req.HostId)
+			hostInv.CurrentAmtState = computev1.AmtState_AMT_STATE_PROVISIONED
+		} else {
+			zlog.Debug().Msgf("Host %s AMT current state is Unprovisioned", req.HostId)
+			hostInv.CurrentAmtState = computev1.AmtState_AMT_STATE_UNPROVISIONED
+		}
+		err = dms.updateHost(ctx, hostInv.GetTenantId(), hostInv.GetResourceId(),
+			&fieldmaskpb.FieldMask{Paths: []string{
+				computev1.HostResourceFieldCurrentAmtState,
+			}}, &computev1.HostResource{
+				CurrentAmtState: hostInv.CurrentAmtState,
+			})
+		if err != nil {
+			zlog.InfraSec().InfraErr(err).Msgf("Failed to update AMT state for host %s", hostInv.GetResourceId())
+			return nil, errors.Errorfc(codes.Internal, "Failed to update AMT state: %v", err)
+		}
+		return &pb.ActivationResultResponse{}, nil
```

Comment:
I dont fully understand this code and there is not status reporting to IN_PROGRESS